shared/datastore.go

```diff
@@ -12,9 +16,40 @@ import (
 	"google.golang.org/appengine/datastore"
 )
 
+// Key abstracts an int64 based datastore.Key
+type Key interface {
+	IntID() int64
+}
+
+// Iterator abstracts a datastore.Iterator
+type Iterator interface {
+	Next(dst interface{}) (Key, error)
+}
+
+// Query abstracts a datastore.Query
+type Query interface {
+	Filter(filterStr string, value interface{}) Query
+	Project(project string) Query
+	Limit(limit int) Query
+	Offset(offset int) Query
+	Order(order string) Query
+	KeysOnly() Query
+	Distinct() Query
+	Run(Datastore) Iterator
+}
+
+// Datastore abstracts a datastore, hiding the distinctions between cloud and
+// appengine's datastores.
+type Datastore interface {
+	Context() context.Context
+	NewQuery(typeName string) Query
+	GetAll(q Query, dst interface{}) ([]Key, error)
```

Comment:
@mdittmer @Hexcles this is where it will be appropriate to follow up by adding methods for fetching runs, and making a mock implementation, to replace the places where we've done that ad-hoc, e.g. 
- https://github.com/web-platform-tests/wpt.fyi/blob/master/api/query/query.go#L26
- https://github.com/web-platform-tests/wpt.fyi/blob/master/api/query/cache/backfill/backfill.go#L24